### ISSUE TITLE
bug in extracting

### ISSUE THREAD

**Issue by micrittenden** [wmgeolab/scope#68]
fix bug where tuple object has no attribute save or nonetype object has no attribute....

**Comment by micrittenden** [comment]
it seems like the bug happens when people try to delete or leave empty the text  box(es) in the middle, while having filled text boxes before and after. For example, I have three extracts for a source, but decided to delete or leave empty the second extract. it will throw an error

**Comment by micrittenden** [comment]
I think i know what the bug is in the case of the nonetype object because I just ran into the same problem in the parsing module. We check to see if extracts already exist, and if not we set it equal to None. So, later, when we try to set the current_status, it is to a nonetype object if it doesn't already exist. however, the form still submits, so if we simply click back then resubmit the form it works (because the object already exists and therefore is not assigned None).
Therefore, we need to set an initial value for the current_status field when we save the form. However, nothing that I have tried has worked for some reason. Im sure it will be a simple fix for you, knowing the above information.
I imagine it is a similar mistake causing the error with the tuple object.
When you manage to fix this, pls let me know so i can make the same change in the parsing_m module

**Comment by micrittenden** [comment]
Ok it was an easy fix for the parsing_m module. I just added the line "activity = Activity.objects.get(extract=extract)" for a second time after the form was saved in the finish=yes section

I have yet to try to resolve the extracting_m bug

**Comment by micrittenden** [comment]
For extracting_m, the bug happens when a user tries to "delete" a form in the formset on the first time they are extracting (i.e., before they have clicked save or finish). "id" for the extract is None because it hasn't been saved in the db before (and thus, it doesn't have an auto-incremented id).
This error is in line 85 of extracting_m/views.py.

I'm unsure how to fix it though.

**Comment by karimbahgat** [comment]
Fixed both issues now, and tested that it works using a variety of user actions. For the None-type bug the fix was as simple as only deleting if the obj was None. For the tuple-error, I had to do some digging to find out that in contrast to deleted_objects and new_objects, changed_objects actually contains 2-tuples (the object itself + a dict of the changed data). There were also some older artifacts in the code that had to be removed. 

See my comments in https://github.com/wmgeolab/scope/commit/afb2facc1922a76800554ee9baf5c4955b0b6b14 for details.
